percona/k8s/util.go

```diff
@@ -32,6 +39,71 @@ func GetWatchNamespace() (string, error) {
 	return ns, nil
 }
 
+func InitContainer(component, image string,
+	pullPolicy corev1.PullPolicy,
+	secCtx *corev1.SecurityContext,
+	resources corev1.ResourceRequirements,
+) corev1.Container {
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      naming.CrunchyBinVolumeName,
+			MountPath: naming.CrunchyBinVolumePath,
+		},
+	}
+
+	return corev1.Container{
+		Name:                     component + "-init",
+		Image:                    image,
+		ImagePullPolicy:          pullPolicy,
+		VolumeMounts:             volumeMounts,
+		Command:                  []string{"/usr/local/bin/init-entrypoint.sh"},
+		TerminationMessagePath:   "/dev/termination-log",
+		TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+		SecurityContext:          secCtx,
+		Resources:                resources,
+	}
+}
+
+func InitImage(ctx context.Context, cl client.Reader, cluster *v1beta1.PostgresCluster) (string, error) {
+	if cluster != nil && len(cluster.Spec.InitImage) > 0 {
+		return cluster.Spec.InitImage, nil
+	}
+	return OperatorImage(ctx, cl)
+}
+
+func OperatorImage(ctx context.Context, cl client.Reader) (string, error) {
```

Comment:
Since this function is only used in this package, we can make it non-exported.

Reply:
https://github.com/percona/percona-postgresql-operator/pull/1098/commits/018718a775e95405ed19092f7d2396fcf01cd7db